### ISSUE TITLE
Error when calling IServiceCollection.AddMvc when also referencing Microsoft.AspNet.WebApi.OwinSelfHost and Nancy.Owin

### ISSUE THREAD

**Issue by christiansparre** [aspnet/Mvc#5425]
Ok, I agree this is a pretty weird issue, but I'm referencing one of our own nuget packages that has a dependency on

```
Microsoft.AspNet.WebApi.OwinSelfHost v5.2.3
Nancy.Owin v1.4.1
```

From a pretty empty test site using ASP.NET Core and MVC

When calling services.AddMvc i'm getting the following

```
An item with the same key has already been added.

   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Microsoft.AspNetCore.Mvc.Internal.DefaultAssemblyPartDiscoveryProvider.CandidateResolver..ctor(IReadOnlyList`1 dependencies, ISet`1 referenceAssemblies)
   at Microsoft.AspNetCore.Mvc.Internal.DefaultAssemblyPartDiscoveryProvider.GetCandidateLibraries(DependencyContext dependencyContext)
   at Microsoft.AspNetCore.Mvc.Internal.DefaultAssemblyPartDiscoveryProvider.GetCandidateAssemblies(Assembly entryAssembly, DependencyContext dependencyContext)
   at Microsoft.AspNetCore.Mvc.Internal.DefaultAssemblyPartDiscoveryProvider.DiscoverAssemblyParts(String entryPointAssemblyName)
   at Microsoft.Extensions.DependencyInjection.MvcCoreServiceCollectionExtensions.GetApplicationPartManager(IServiceCollection services)
   at Microsoft.Extensions.DependencyInjection.MvcCoreServiceCollectionExtensions.AddMvcCore(IServiceCollection services)
   at Microsoft.Extensions.DependencyInjection.MvcServiceCollectionExtensions.AddMvc(IServiceCollection services)
```

Heres an example of the error: https://github.com/christiansparre/AddMvcError 

Is there a way to configure MVC not to try to manually discover the "Parts"? Searched a bit but did not really find anything. I'm guessing the Nancy/WebAPI combination have something named the same as something inside ASP.NET Core and that causes a conflict.

Hope someone can help with this error :)

**Comment by rynowak** [comment]
@christiansparre - try manually registering the `ApplicationPartManager` as a service before `AddMvc()`

This is what we do in our functional tests to prevent each from reaching outside its assembly. https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.FunctionalTests/MvcTestFixture.cs#L68

**Comment by christiansparre** [comment]
@rynowak Thanks, that worked :)

What providers should be added as default for MVC to work properly?

**Comment by Eilon** [comment]
@javiercn do you have any thoughts on this?

**Comment by javiercn** [comment]
@Eilon I will take a look at it.

**Comment by Eilon** [comment]
Moving to 1.2.0.

**Comment by balchen** [comment]
We saw this issue today as well, only in different circumstances.

Our ASP.NET Core project references a portable library which in turn references Newtonsoft.Json.
Last week, in an attempt to resolve platform/build issues, I changed the portable library to a .NET Standard library.

After this change, we started seeing the error message from the inital issue description, and the proposed solution worked fine for us.

The only non-platform reference in the portable/.NET Standard library is Newtonsoft.Json.

Just FYI.

**Comment by rynowak** [comment]
Load balancing this, @jbagga can you see what's the problem here?

**Comment by jbagga** [comment]
This seems like a repro for https://github.com/aspnet/Mvc/issues/5289. It consistently throws since `Nancy.Owin` 1.4.1 uses all uppercase for its dependency on Owin (note that 2.0.0-alpha and above use "Owin" not "OWIN" so it is not a problem there) when using ASP.NET Core 1.x.

With 2.0, this cannot be reproduced as Owin and OWIN are normalized into one package dependency.

cc @Eilon

**Comment by Eilon** [comment]
I think given that the bug is ultimately outside of ASP.NET Core, and that it's already fixed in 2.0, it should be ok to close this.
